README
NAME
    Nitesi::DBI - DBI bundle for Nitesi shop machine

VERSION
    Version 0.0091

DESCRIPTION
    This bundle contains:

        Nitesi::Query::DBI

        Nitesi::Account::Provider::DBI

AUTHOR
    Stefan Hornburg (Racke), `<racke at linuxia.de>'

BUGS
    Please report any bugs or feature requests to `bug-nitesi-dbi at
    rt.cpan.org', or through the web interface at
    http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Nitesi-DBI. I will be
    notified, and then you'll automatically be notified of progress on your
    bug as I make changes.

SUPPORT
    You can find documentation for this module with the perldoc command.

        perldoc Nitesi::DBI

    You can also look for information at:

    * RT: CPAN's request tracker (report bugs here)
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=Nitesi-DBI

    * AnnoCPAN: Annotated CPAN documentation
        http://annocpan.org/dist/Nitesi-DBI

    * CPAN Ratings
        http://cpanratings.perl.org/d/Nitesi-DBI

    * Search CPAN
        http://search.cpan.org/dist/Nitesi-DBI/

LICENSE AND COPYRIGHT
    Copyright 2011-2013 Stefan Hornburg (Racke).

    This program is free software; you can redistribute it and/or modify it
    under the terms of either: the GNU General Public License as published
    by the Free Software Foundation; or the Artistic License.

    See http://dev.perl.org/licenses/ for more information.



Changes
Revision history for Nitesi::DBI

0.0091 Wed Aug 21 11:48:44 2013 CEST

    [BUG FIXES]

    * Avoid trying to load entities without key in DBI backend.
    * Check whether there is a base role when handling subclasses for
      assignments.
       
    [DOCUMENTATION]

    * Add session_id to carts table in database schema.
    * Get rid of "::character varying" as default value in the database schema.
    * Fix mistakes in definition of carts table in database schema.

0.0090 Thu Jun 20 09:22:36 2013 CEST

    [API CHANGES]

    * Use integer type for last_login field in users.

    [ENHANCEMENTS]

    * Replace Nitesi::Object with Moo.
    * Add support for last_login to account provider.

0.0082 Sun Jun 16 13:13:09 2013 CEST

    [BUG FIXES]

    * Exclude SQLite2 DBI driver from limit_offset and serial tests.

0.0081 Tue Jun 11 17:35:19 2013 CEST

    [ENHANCEMENTS]

    * Add support for offset keyword in select queries (Grega Pompe).
    * Add log_queries attribute to Dancer::Plugin::Nitesi::Backend::DBI.

    [DOCUMENTATION]

    * Add gtin and status columns to products table in the database schema.

    [TESTS]

    * Fix regular expression for tables to be cleared in query test script.

0.0080 Sun May 26 11:26:16 2013 CEST

    [ENHANCEMENTS]

    * Add Git repository to Makefile.PL.

    [BUG FIXES]

    * Execute POD tests only during release testing.

    [DOCUMENTATION]

    * Increase size of username field to the size of the email field.
    * Update schema in regards of transactions, orderline and 
      payment_transactions.
    * Add canonical_sku and uri column to products table in the database schema.

    [TESTS]

    * Fix detection of existing tables for PostgreSQL in the query test.

0.0060 Sun Oct 28 18:20:49 2012 CET

    [ENHANCEMENTS]

    * Add DBI backend for Nitesi API.
    * Add DBI backend class for Dancer.

    [DOCUMENTATION]

    * Improve documentation with more examples and explanations.
      (Jure Kodžoman)
    * Updates to draft on recommended database schema.

0.0050 Wed Jul 11 14:10:27 2012 CEST

    [ENHANCEMENTS]

    * Let insert method try to return value of primary key. 
      With mysql and Pg drivers that will work even with serial fields.
    * Add load, create, delete and become method to account provider class.

    [BUG FIXES]

    * Fix bug in Nitesi::Query::DBI's delete method when called with named
      parameters.

    [TESTS]

    * Add tests for Nitesi::Query::DBI's delete method.
    * Remove existing tables from test database at the beginning of
      the test cycle.

    [DOCUMENTATION]

    * Add draft documentation on Nitesi's default database schema.
    * Add example for calling Nitesi::Query::DBI's delete method with named
      parameters. 

0.0040 Sun Nov 27 12:14:54 2011 CET

    [ENHANCEMENTS]

    * Add support for distinct in select queries to query engine.
    * Add inactive setting for disabled users to account provider.
    * Add log_queries attribute to Nitesi::Query::DBI class.

    [BUG FIXES]

    * Fix crash with select queries passed to SQL::Abstract::More without where parameter.
    * Fix create and drop table statement

    [TESTS]

    * Add tests for select distinct queries.

    [DOCUMENTATION]

    * Add supported query types and examples for select queries with order, limit,
      distinct and join to Nitesi::Query::DBI's POD.

0.0032 Mon Nov  7 08:49:29 2011 CET

    [BUG FIXES]

    * Require SQL::Statement for query testing as DBI::SQL::Nano is
      too limited for the statements produced by SQL::Abstract.

0.0031 Sun Oct 30 10:38:40 2011 CET

    [TESTS]

    * Remove -T switch from simple query test.

    [DOCUMENTATION]

    * Add documentation for dbh and sqla accessors.

0.0030 Sat Oct 29 13:21:28 2011 CEST

    [BUG FIXES]

    * Skip query tests when database handles aren't available.

0.0020 Wed Oct 26 16:28:03 2011 CEST

    [API CHANGES]

    * Change semantic of roles account method to return role names
      instead of role identifiers.

    [ENHANCEMENTS]

    * Extend value method of DBI account provider with setting account values.
    * Wrap calls to SQL::Abstract(::More)'s select into an eval.

    [BUG FIXES]

    * Allow field as parameter to select_list_field query method analogous to select_field.

    [DOCUMENTATION]

    * Use =head1 for description in Nitesi::Query::DBI's POD.
    * Add description to POD for DBI provider's exists method.

    [TESTS]

    * Add very basic test for Nitesi::Query::DBI.

0.0010 Thu Oct 13 09:10:59 2011 EDT

    [ENHANCEMENTS]

    * Add exists and password method to account provider class.
    * Add fields attribute to account provider class.

    [BUG FIXES]

    * Add dependency on Nitesi to Makefile.PL.

    [DOCUMENTATION]

    * Fix code example for select_field method.

0.0003 Tue Oct  4 11:50:55 2011 EDT

    [ENHANCEMENTS]

    * Add limit parameter to select method.

0.0002 Thu Sep 29 13:54:10 2011 CEST

    [ENHANCEMENTS]

    * Return all fields from a select query without fields parameter.
    * Pass through order parameter for select queries.

0.0001 Sat Sep 24 15:49:04 2011 CEST

    * Initial release




lib/Nitesi/DBI.pm
package Nitesi::DBI;

use 5.006;
use strict;
use warnings;

=head1 NAME

Nitesi::DBI - DBI bundle for Nitesi shop machine

=head1 VERSION

Version 0.0091

=cut

our $VERSION = '0.0091';

=head1 DESCRIPTION

This bundle contains:

=over 4

=item 

L<Nitesi::Query::DBI>

=item

L<Nitesi::Account::Provider::DBI>

=back

=head1 AUTHOR

Stefan Hornburg (Racke), C<< <racke at linuxia.de> >>

=head1 BUGS

Please report any bugs or feature requests to C<bug-nitesi-dbi at rt.cpan.org>, or through
the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Nitesi-DBI>.  I will be notified, and then you'll
automatically be notified of progress on your bug as I make changes.

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Nitesi::DBI


You can also look for information at:

=over 4

=item * RT: CPAN's request tracker (report bugs here)

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=Nitesi-DBI>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/Nitesi-DBI>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/Nitesi-DBI>

=item * Search CPAN

L<http://search.cpan.org/dist/Nitesi-DBI/>

=back

=head1 LICENSE AND COPYRIGHT

Copyright 2011-2013 Stefan Hornburg (Racke).

This program is free software; you can redistribute it and/or modify it
under the terms of either: the GNU General Public License as published
by the Free Software Foundation; or the Artistic License.

See http://dev.perl.org/licenses/ for more information.


=cut

1; # End of Nitesi::DBI


Makefile.PL
use 5.006;
use strict;
use warnings;
use ExtUtils::MakeMaker;

WriteMakefile(
    NAME                => 'Nitesi::DBI',
    AUTHOR              => q{Stefan Hornburg (Racke) <racke@linuxia.de>},
    VERSION_FROM        => 'lib/Nitesi/DBI.pm',
    ABSTRACT_FROM       => 'lib/Nitesi/DBI.pm',
    ($ExtUtils::MakeMaker::VERSION >= 6.3002
      ? ('LICENSE'=> 'perl')
      : ()),
    PL_FILES            => {},
    BUILD_REQUIRES => {
        'Test::More' => 0,
	'Test::Database' => 0,
	'SQL::Statement' => 0,
    },
    PREREQ_PM => {
	'SQL::Abstract' => 0,
	'SQL::Abstract::More' => 0,
        'Nitesi' => 0.0090,
    },
    dist                => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
    clean               => { FILES => 'Nitesi-DBI-*' },
    META_MERGE        => {
        resources => {
            repository  => 'https://github.com/racke/Nitesi-DBI',
        },
    },
);
